pulsar-client/src/test/java/org/apache/pulsar/client/impl/BuildersTest.java

```diff
@@ -69,4 +75,33 @@ public void enableTlsTest() {
         assertEquals(builder.conf.isUseTls(), false);
         assertEquals(builder.conf.getServiceUrl(), "pulsar+ssl://service:6650");
     }
+
+    @Test
+    public void readerBuilderLoadConfTest() {
+        try (PulsarClient client = PulsarClient.builder().serviceUrl("pulsar://localhost:6650").build()) {
+            String topicName = "test_src";
+            MessageId messageId = new MessageIdImpl(1, 2, 3);
+            Map<String, Object> config = new HashMap<>();
+            config.put("topicName", topicName);
+            config.put("receiverQueueSize", 2000);
+            ReaderBuilderImpl<byte[]> builder = (ReaderBuilderImpl<byte[]>) client.newReader()
+                .startMessageId(messageId)
+                .loadConf(config);
+
+            Class<?> clazz = builder.getClass();
+            try {
+                Field conf = clazz.getDeclaredField("conf");
+                Object obj = conf.get(clazz);
+                assertTrue(obj instanceof ReaderConfigurationData);
+                if (obj instanceof ReaderConfigurationData) {
+                    assertEquals(((ReaderConfigurationData) obj).getTopicName(), topicName);
+                    assertEquals(((ReaderConfigurationData) obj).getStartMessageId(), messageId);
+                }
+            } catch (NoSuchFieldException | IllegalAccessException e) {
+
+            }
+        } catch (PulsarClientException e) {
+            assertTrue(e instanceof PulsarClientException);
```

Comment:
e is already a PulsarClientException. I am not sure why you add this assertion here.